lib/Service/SlackAPIService.php

```diff
@@ -118,21 +121,46 @@ private function getUserRealName(string $userId, string $slackUserId): ?string {
 
 	/**
 	 * @param string $userId
+	 * @param bool $useCache
 	 * @return array
 	 * @throws PreConditionNotMetException
 	 */
-	public function getMyChannels(string $userId): array {
-		$channelResult = $this->request($userId, 'conversations.list', [
-			'exclude_archived' => true,
-			'types' => 'public_channel,private_channel,im,mpim'
-		]);
-
-		if (isset($channelResult['error'])) {
-			return (array)$channelResult;
+	public function getMyChannels(string $userId, bool $useCache): array {
+		$dCache = $this->cacheFactory->createDistributed(Application::APP_ID);
+		$cacheKey = 'channels-' . $userId;
+
+		if (!$useCache) {
+			$dCache->remove($cacheKey);
+		} elseif ($cachedChannels = $dCache->get($cacheKey)) {
+			$this->logger->debug('Slack channels cache hit', ['userId' => $userId]);
+			return $cachedChannels;
 		}
 
-		if (!isset($channelResult['channels']) || !is_array($channelResult['channels'])) {
-			return ['error' => 'No channels found'];
+		$cursor = 'dummdumm'; // initial value
+		$rawChannels = [];
+
+		while ($cursor !== '') {
```

Comment:
We could try to make this a little bit safer:
* stop the loop if there are more than 10000 raw channels (or another relevant number)
* stop the loop if we think it is infinite: if the next_cursor is equal to the current cursor

Reply:
Yeah and the MAX_CHANNELS_TO_FETCH will stop potential infinite loops anyway.